FirebasePerformance/Sources/AppActivity/FPRAppActivityTracker.m

```diff
@@ -147,6 +160,38 @@ - (FIRTrace *)activeTrace {
   return self.backgroundSessionTrace;
 }
 
+- (void)startTrackingNetwork {
+  self.networkType = firebase_perf_v1_NetworkConnectionInfo_NetworkType_NONE;
+
+#if TARGET_OS_IOS
+  if (@available(iOS 12, *)) {
+#elif TARGET_OS_TVOS
+  if (@available(tvOS 12, *)) {
+#endif
+    dispatch_queue_attr_t attrs = dispatch_queue_attr_make_with_qos_class(
+        DISPATCH_QUEUE_SERIAL, QOS_CLASS_UTILITY, DISPATCH_QUEUE_PRIORITY_DEFAULT);
+    self.monitorQueue = dispatch_queue_create("com.google.perf.network.monitor", attrs);
+
+    self.monitor = nw_path_monitor_create();
+    nw_path_monitor_set_queue(self.monitor, self.monitorQueue);
+    nw_path_monitor_set_update_handler(self.monitor, ^(nw_path_t _Nonnull path) {
+      BOOL isWiFi = nw_path_uses_interface_type(path, nw_interface_type_wifi);
```

Comment:
I see under completion block we are mutating the attributes value for self, will it be safer to use weakSelf to do so?

Reply:
Excellent catch. Thanks for keeping my honest. Been a while catching these patterns in ObjC. Made the change.